Assets/Scripts/NetworkStatusManager.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting;
+using UnityEditor.EditorTools;
+using UnityEngine;
+
+public class NetworkStatusManager : MonoBehaviour
```

Comment:
Naming: how about something like StatusDisplayHelper? I can imagine us re-using this class for other important app states besides just network. And I would argue this doesn't actually manage our network status; it just displays it.